### PR TITLE
fix: issue in the event location field

### LINKED ISSUE #15225: [CAL-3916] bug: issue in the event location field
Found a bug? Please fill out the sections below. 👍

### Issue Summary

After changing to any other conference app location in managedEvent, the same link is present in the input field as in the previous app.

### Steps to Reproduce

1. Select a conference app for the event location that has a static link and save it after setting the link.
2. Change to any other conference app with a static link type.
3. The previous link is still present in the input field.

Evidence:-

Screenshot 2024-05-29 003719

<sub>[CAL-3916](https://linear.app/calcom/issue/CAL-3916/bug-issue-in-the-event-location-field)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #15225 
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Nice work @anikdhabal ! A small problem tho.

If I select a location with static link, save it, and then change it to use locations without static link, it doesn't let me save. There are no visible errors or anything, the button just does not press. I have to delete that location, and add a new one to make it work.


https://github.com/calcom/cal.com/assets/74371312/63ca4ea0-62fa-4399-bbb0-824f3d287204

File: apps/web/components/eventtype/EventSetupTab.tsx [hunk 1]
```diff
@@ -275,6 +275,11 @@ export const EventSetupTab = (
                               teamName: e.teamName,
                             }),
                           });
+                          const newDefaultValue = eventLocationType.defaultValueVariable ? "" : undefined;
```
Comment (Amit91848) [inline apps/web/components/eventtype/EventSetupTab.tsx:278]: As per my understanding all integrations have some sort of defaultValueVariable, I don't think this is the right way to fix this.

**Comment by anikdhabal** [comment]
So the issue is that if there is a location with link as the default variable and a link is saved, when selecting any other location with link as the default variable, the form value or link does not change. This is expected because the locationInput field name depends on the index and default variable. Therefore, if we select any location at that index number with link type, the value does not change.

So, to fix this, we need to update the locationInput value, means need to set locationField link to an empty string under certain conditions where link is the default variable. I've fixed that.

**Review by Amit91848 — APPROVED** [review]
LGTM! Tested and works great. 

Will wait for another approval from @Udit-takkar

**Review by ibex088 — APPROVED** [review]

**Review by Udit-takkar — COMMENTED** [review]
@anikdhabal  So the issue is that `updateLocationField` only updates the properties that you pass in the second parameter and doesn't remove the properties from previously selected location type. 

I think we should also pass other properties as null/empty string. right now you only passed the `link` property. there are more properties like this. what do you think?

**Comment by anikdhabal** [comment]
> @anikdhabal So the issue is that `updateLocationField` only updates the properties that you pass in the second parameter and doesn't remove the properties from previously selected location type.
> 
> I think we should also pass other properties as null/empty string. right now you only passed the `link` property. there are more properties like this. what do you think?

Yeah that make sense. Let me update the pr

**Comment by anikdhabal** [comment]
Hey @Udit-takkar, not really needed. Actually, the problem arises when a similar default variable location is selected. As you can see in locationInput, the props name depends on the default variable. So, if two similar locations with default variables are selected consecutively, the form props value does not change. We mainly need two things to change and save a location: one is the type, which is already updated, and the other is the value, if any. The main issue occurs when the location default variable is set to 'link'. Since this has many options and we need to enter an input, I think it's okay to set the link or input value for the link type empty

**Pull request merged**